app/root.tsx

```diff
@@ -1,14 +1,33 @@
-import { Links, Link, Meta, Outlet, Scripts, ScrollRestoration } from '@remix-run/react';
-import '~/styles/index.scss';
+import {
+    Links,
+    Link,
+    Meta,
+    Outlet,
+    Scripts,
+    ScrollRestoration,
+    useLoaderData,
+} from '@remix-run/react';
+import { json } from '@remix-run/node';
 import { SiteWrapper } from '~/components/site-wrapper/site-wrapper';
 import { ROUTES } from '~/router/config';
-import { RouteHandle } from '@remix-run/react/dist/routeModules';
+import { RouteHandle } from '~/router/types';
+import { EcomAPIContextProvider } from '~/api/ecom-api-context-provider';
+import { CartOpenContextProvider } from '~/components/cart/cart-open-context';
+import '~/styles/index.scss';
+
+export async function loader() {
+    return json({
+        ENV: {
+            WIX_CLIENT_ID: process?.env?.WIX_CLIENT_ID,
+        },
+    });
+}
 
 export const handle: RouteHandle = {
     breadcrumb: () => <Link to={ROUTES.home.path}>Home</Link>,
 };
 
-export default function App() {
+export function Layout({ children }: React.PropsWithChildren) {
```

Comment:
`React.PropsWithChildren` - I didn't know, cool.

Reply:
Where is this `Layout` component used?

Reply:
This is a Remix-specific thing https://remix.run/docs/en/main/file-conventions/root#layout-export